migrations/migrations/0001_archive_meetings.py

```diff
@@ -0,0 +1,113 @@
+
+from datastore.migrations import (
+    BaseEvent,
+    BaseMigration,
+    CreateEvent,
+    DeleteEvent,
+    ListUpdateEvent,
+    MigrationKeyframeAccessor,
+    PositionData,
+    RestoreEvent,
+)
+from datastore.shared.util import collection_and_id_from_fqid
+
+
+class Migration(BaseMigration):
+    """
+    This migration does:
+    - Add `organization/limit_of_meetings`. Default 0 (no limit).
+    - Add the 1:N relation `organization/active_meeting_ids` <-> `meeting/is_active_in_organization_id`.
+      This relation must be set for every meeting and link them to the one organization.
+
+    Note that the field `organization/active_meeting_ids` is modified at the end of the migration. Why?
+    Remember the single assertion about consistency within the datastore: The data is consistent *after*
+    each position. When migrating single events, the consistency is not (necessarily) provided.
+    One seemingly simpler but wrong migration would be:
+
+    ```
+    if collection == "meeting":
+        if isinstance(event, CreateEvent):
+            event.data["is_active_in_organization_id"] = 1
+            update_event = ListUpdateEvent("organization/1", {"add": {"active_meeting_ids": [id]})
+            new_event = [event, update_event]
+        elif isinstance(event, DeleteEvent):
+            update_event = ListUpdateEvent("organization/1", {"remove": {"active_meeting_ids": [id]})
+            new_event = [event, update_event]
+        elif isinstance(event, RestoreEvent):
+            update_event = ListUpdateEvent("organization/1", {"add": {"active_meeting_ids": [id]})
+            new_event = [event, update_event]
+    ```
+
+    The problem is, that within the events of the one position it is not clear whether the organization exists!
+    The organization itself might be created with the last event, so doing a ListUpdateEvent before the creation
+    of the organization will fail. This is the reason `get_additional_events` is used. After migrating all events
+    we can be sure, that the content is consistent (with the exception of `meeting/is_active_in_organization_id
+    since we added just half of the relation). So when appending to the events we can be sure that the
+    organization exists.
+
+    Also note that no other event than create is handled for organization since deleting (so restoring, too) is
+    not yet supported and since we added two new fields they are cannot be affected in any update event.
+    """
+
+    target_migration_index = 2
+
+    def position_init(self) -> None:
+        # Capture all meeting ids to add/remove from
+        # `organization/active_meeting_ids` in this position.
+        self.meeting_ids_to_add: Set[int] = set()
+        self.meeting_ids_to_remove: Set[int] = set()
+
+    def migrate_event(
+        self,
+        event: BaseEvent,
+        old_accessor: MigrationKeyframeAccessor,
+        new_accessor: MigrationKeyframeAccessor,
+        position_data: PositionData,
```

Comment:
Just saw https://github.com/OpenSlides/openslides-datastore-service/pull/146, is the datastore not up-to-date in the backend?

Reply:
Ok, I adjusted mypy a bit. Now, it is checked with the datastore together.

```
Argument 2 to "ListUpdateEvent" has incompatible type "Dict[str, Dict[str, List[int]]]"; expected "Dict[str, Dict[Union[str, _Field], List[Union[str, int]]]]"
```
IDK whats wrong here, I'll use `Any` (what I know is that I do not like mypy :D)

Reply:
Yeah, mypy seems to have problems with nested unions sometimes... Did you try `Dict[str, ListUpdatesDict]`, like defined in the datastore?